featurebyte/routes/common/base_table.py

```diff
@@ -258,6 +258,16 @@ async def update_column_entity(
         TableDocumentT
             Table object with updated entity
         """
+        document = await self.service.get_document(document_id=document_id)
+
+        col_info = [col for col in document.columns_info if col.name == column_name]
+        if col_info and col_info[0].semantic_id:
+            semantic = await self.semantic_service.get_document(document_id=col_info[0].semantic_id)
+            if semantic and semantic.name == SemanticType.SCD_SURROGATE_KEY_ID.value:
+                raise EntityTaggingIsNotAllowedError(
+                    f"Column {column_name} cannot be tagged with entity {entity_id}"
```

Comment:
Would it be useful to make the error message more specific, such as `Surrogate key column {column_name} cannot be tagged with entity {entity_id}`?